codeview/src/main/java/io/github/kbiakov/codeview/adapters/AbstractCodeAdapter.kt

```diff
@@ -67,52 +56,71 @@ abstract class AbstractCodeAdapter<T> : RecyclerView.Adapter<AbstractCodeAdapter
      */
     constructor(context: Context,
                 content: String,
+                theme: ColorThemeData,
                 isShowFull: Boolean = true,
                 maxLines: Int = MAX_SHORTCUT_LINES,
                 shortcutNote: String = context.getString(R.string.show_all),
                 listener: OnCodeLineClickListener? = null) {
-        mContext = context
-        mContent = content
-        mMaxLines = maxLines
-        codeListener = listener
-
-        if (isShowFull) {
-            isFullShowing = true
-            mLines = extractLines(content)
-        } else
-            initCodeContent(isShowFull, shortcutNote)
+        this.context = context
+
+        highlighter = Highlighter(context)
+
+        highlighter.code = content
+        highlighter.maxLines = maxLines
+        highlighter.lineClickListener = listener
+        highlighter.theme = theme
+        highlighter.shortcut = !isShowFull
+        highlighter.shortcutNote = shortcutNote
+
+        prepareCodeLines()
     }
 
+    fun isShorted(): Boolean = droppedLines != null
+
+    //todo: showAllNotes()
```

Comment:
If you want to implement this features, implement them in this PR. Otherwise, please, move all TODO's to separated file. You can create ```CONTRIBUTING.md``` with all necessary updates. Also, you can cut & paste this section from ```README.md```.